dotnet/src/Connectors/Connectors.Memory.SqliteVec/SqliteServiceCollectionExtensions.cs

```diff
@@ -1,84 +1,227 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
-using Microsoft.Data.Sqlite;
+using System;
 using Microsoft.Extensions.AI;
-using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel;
 using Microsoft.SemanticKernel.Connectors.SqliteVec;
 
-namespace Microsoft.SemanticKernel;
+namespace Microsoft.Extensions.DependencyInjection;
 
 /// <summary>
 /// Extension methods to register SQLite <see cref="VectorStore"/> instances on an <see cref="IServiceCollection"/>.
 /// </summary>
 public static class SqliteServiceCollectionExtensions
 {
     /// <summary>
-    /// Register a SQLite <see cref="VectorStore"/> with the specified service ID.
-    /// <see cref="SqliteConnection"/> instance will be initialized, connection will be opened and vector search extension with be loaded.
+    /// Registers a <see cref="SqliteVectorStore"/> as <see cref="VectorStore"/>, with the specified connection string and service lifetime.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="VectorStore"/> on.</param>
-    /// <param name="connectionString">Connection string for <see cref="SqliteConnection"/>.</param>
-    /// <param name="options">Optional options to further configure the <see cref="VectorStore"/>.</param>
-    /// <param name="serviceId">An optional service id to use as the service key.</param>
-    /// <returns>Service collection.</returns>
+    /// <inheritdoc cref="AddVectorStore"/>
     public static IServiceCollection AddSqliteVectorStore(
         this IServiceCollection services,
-        string connectionString,
-        SqliteVectorStoreOptions? options = default,
-        string? serviceId = default)
-        => services.AddKeyedSingleton<VectorStore>(
-            serviceId,
-            (sp, _) => new SqliteVectorStore(connectionString, options ?? sp.GetService<SqliteVectorStoreOptions>() ?? new() { EmbeddingGenerator = sp.GetService<IEmbeddingGenerator>() }));
+        Func<IServiceProvider, string> connectionStringProvider,
```

Comment:
Unless I missed it, I didn't see a public AddPostgresVectorStore that accepts Funcs